File: medical-app/src/app/services/video-call-message.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface VideoCallMessageDto {
  id?: number;
  roomId: string;
  senderName: string;
  senderId?: number;
  senderType?: string;
  message: string;
  createdAt?: string;
}

@Injectable({
  providedIn: 'root',
})
export class VideoCallMessageService {
  private baseUrl = 'http://localhost:8080/api/video-call-messages';

  constructor(private http: HttpClient) {}

  private getHeaders(): HttpHeaders {
    const token = localStorage.getItem('token');
    return new HttpHeaders({
      'Authorization': `Bearer ${token}`,
      'Content-Type': 'application/json'
    });
  }

  /**
   * Save a single chat message
   */
  saveMessage(message: VideoCallMessageDto): Observable<VideoCallMessageDto> {
    return this.http.post<VideoCallMessageDto>(this.baseUrl, message, {
      headers: this.getHeaders()
    });
  }

  /**
   * Save multiple messages at once (batch save at end of call)
   */
  saveMessages(messages: VideoCallMessageDto[]): Observable<VideoCallMessageDto[]> {
    return this.http.post<VideoCallMessageDto[]>(`${this.baseUrl}/batch`, messages, {
      headers: this.getHeaders()
    });
  }

  /**
   * Get all messages for a specific room/call
   */
  getMessagesByRoom(roomId: string): Observable<VideoCallMessageDto[]> {
    return this.http.get<VideoCallMessageDto[]>(`${this.baseUrl}/room/${roomId}`, {
      headers: this.getHeaders()
    });
  }

  /**
   * Get rooms that a user has participated in
   */
  getRoomsForUser(userId: number): Observable<string[]> {
    return this.http.get<string[]>(`${this.baseUrl}/user/${userId}/rooms`, {
      headers: this.getHeaders()
    });
  }
}

File: medical-you/medical-you/src/main/java/com/zega/medical_you_be/service/VideoCallMessageService.java
package com.zega.medical_you_be.service;

import com.zega.medical_you_be.model.dto.VideoCallMessageDto;

import java.util.List;

/**
 * Service for managing video call chat messages.
 */
public interface VideoCallMessageService {

    /**
     * Save a new chat message.
     */
    VideoCallMessageDto saveMessage(VideoCallMessageDto messageDto);

    /**
     * Save multiple messages at once (for batch save at end of call).
     */
    List<VideoCallMessageDto> saveMessages(List<VideoCallMessageDto> messages);

    /**
     * Get all messages for a specific room/call.
     */
    List<VideoCallMessageDto> getMessagesByRoomId(String roomId);

    /**
     * Get rooms that a user has participated in.
     */
    List<String> getRoomsForUser(Long userId);
}
